cmd/soroban-cli/src/commands/contract/invoke.rs

```diff
@@ -260,6 +268,13 @@ impl Cmd {
             eprintln!("Contract auth: {}", serde_json::to_string(&auth).unwrap());
         }
 
+        if self.events {
+            eprintln!(
+                "Simulated events: {}",
+                serde_json::to_string(&events).unwrap()
```

Comment:
Also, should it always print the simulated events? or should it prefer to print the real result events, and only print the simulated events if the simulation fails?

Reply:
> Seems like this wouldn't pretty-print them, just dump them as an array, right?

Note that events is an unmarshalled DiagnosticEvents vector. I assumed serde would do the right thing and pretty-print them (like I assume happens with `auth` and `footprint`)

> Also, should it always print the simulated events?

Uhm, we should also print them if the transaction submission fails otherwise there wouldn't be any events.

I decided that I didn't want to complicate the logic too much to make it more predictable.

Reply:
Right, I'm saying:
```
if simulation_failed {
  // We return early in this case, so we should print the events here.
  print(simulation.events)
  return error;
}

// But normally we care about the real events, not the simulated.
let real_result = ...;
print(real_result.events);
```

Reply:
Right, I'm saying:
```rs
if simulation_failed {
  // We return early in this case, so we should print the events here.
  print(simulation.events);
  return error;
}

// But normally we care about the real events, not the simulated.
let real_result = ...;
print(real_result.events);
```

Reply:
What if simulation works but transaction submission fails, wouldn't you want the simulation events printed out?

Reply:
@paulbellamy Looking at the preflight library code, in practice you can't really get events if the simulation fails https://github.com/stellar/soroban-tools/blob/main/cmd/soroban-rpc/lib/preflight/src/lib.rs#L192-L202